pandas/io/sql.py

```diff
@@ -564,6 +564,10 @@ def to_sql(frame, name, con, flavor='sqlite', schema=None, if_exists='fail',
         raise NotImplementedError("'frame' argument should be either a "
                                   "Series or a DataFrame")
 
+    if dtype and not isinstance(dtype,dict):
+        temp_type = dtype
+        dtype = { col_name : temp_type for col_name in frame }
```

Comment:
I think flake8 will complain about this line. Need to get rid of the spaces around the curly braces.
Oh, also space after the common above in the `isinstance` check